conda_env/env.py

```diff
@@ -58,11 +60,81 @@ def from_yaml(yamlstr, **kwargs):
     return Environment(**data)
 
 
-def from_file(filename):
+def custom_selectors_from_yaml(yamlstr):
+    """Load and return a list of custom selectors from a given ``yaml string``"""
+    data = yaml.load(yamlstr)
+    if 'selectors' in data:
+        return data['selectors']
+    return []
+
+
+def from_file(filename, selectors=None):
     if not os.path.exists(filename):
         raise exceptions.EnvironmentFileNotFound(filename)
-    with open(filename, 'rb') as fp:
-        return from_yaml(fp.read(), filename=filename)
+    with open(filename, 'r') as fp:
+        yamlstr = fp.read()
+        custom_selectors = custom_selectors_from_yaml(yamlstr)
+        yamlstr = select_lines(yamlstr, filename, custom_selectors, selectors=selectors)
+        return from_yaml(yamlstr, filename=filename)
+
+
+def ns_cfg(custom_selectors, selectors=None):
+    plat = sys.platform  # see https://docs.python.org/2/library/sys.html#sys.platform
+    is_x64 = sys.maxsize > 2**32  # from https://docs.python.org/2/library/platform.html#cross-platform
+    d = dict(
+        linux=plat.startswith('linux'),
+        linux32=plat.startswith('linux') and not is_x64,
+        linux64=plat.startswith('linux') and is_x64,
+        osx=plat.startswith('darwin'),
+        win=plat.startswith('win32'),
+        win32=plat.startswith('win32') and not is_x64,
+        win64=plat.startswith('win32') and is_x64,
+        unix=plat.startswith(('linux', 'darwin')),
+        unix32=plat.startswith(('linux', 'darwin')) and not is_x64,
+        unix64=plat.startswith(('linux', 'darwin')) and is_x64,
+        x86=not is_x64,
+        x64=is_x64,
+        os=os,
+        environ=os.environ,
+    )
+    if selectors is None:
+        selectors = []
+    selector_dict = dict((s, s in selectors) for s in custom_selectors)
+    d.update(selector_dict)
+    return d
+
+
+sel_pat = re.compile(r'(.+?)\s*(#.*)?\[(.+)\](?(2).*)$')
```

Comment:
Yeah, @tadeu mentioned it to me in person. I agree with you that it seems to be supported only for legacy reasons.

Since this is a new feature, I would rather to be a little more strict and always require `#` here, but that's just my opinion.